### PR TITLE
SpriteSheet demo

### PR DESCRIPTION
**Pull request by NigelBreslaw**
This is for after 1.8. Earlier version was driven by a Timer. This one just uses bindings. It is harder to understand though. Maybe it could be simplified?

### CODE REVIEW

**Comment by ogoffart** [comment]
I tried in on this link: https://snapshots.slint.dev/master/editor/?load_url=https://github.com/slint-ui/slint/blob/89d717e2bf31ac7c6d8e8773eac73a8db31fd390/examples/sprite_sheet/SpriteSheet_Demo.slint
Looks nice! but the bounding on the left and right is a bit strange: I think it bounces too early (much before the edge), and the bouncing looks unnatural.

For the efte, i can get much better result with  
```slint
   width: sheet.width;
   height: sheet.height;
```
in SpriteSheet.slint

I wonder if it is needed to change the rotating direction. `frame: self.totalFrames * ( animation-tick() / 400ms);` instead of adjusting with the direction of travel already remove some of the "cut".


Anyway, regarding naming. I think it would be better to use dashes instead of underscores.
Also could be worth making the demo self contained in only one file. What do you thik?

**Comment by NigelBreslaw** [comment]
How do those changes look?

File: examples/sprite_sheet/demo.slint [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: MIT
+
+import { ComboBox } from "std-widgets.slint";
+import { SpriteSheet } from "SpriteSheet.slint";
+
+enum TravelDirection { LEFT, RIGHT }
```
Comment (ogoffart) [inline examples/sprite_sheet/demo.slint:7]: we usually use lower case for enum values

```suggestion
enum TravelDirection { left, right }
```

**Review by ogoffart — COMMENTED** [review]
I think i'd do it like that instread of using changed event:


```slint
SpriteSheet {
  property <int> frame-base;
  frame: frame-base + (travel-direction == TravelDirection.LEFT ? 1 : -1) * animation-tick() / 16ms;
  
   updateX() {
     //...
       travel-direction = ...
       frame-base = self.frame;
   }
```


Also sprite_sheet => sprite-sheet

File: examples/sprite_sheet/demo.slint [hunk 2]
```diff
@@ -50,17 +51,12 @@ export component AppWindow inherits Window {
         x: 0px;
         animate x { duration: 3s; }
         y: (-400px * abs(sin(360deg * animation-tick() / 3s))) + parent.height - ball.height;
-
+        frame: frame-base + (travel-direction == TravelDirection.left ? 1 : -1) * animation-tick() / 16ms;
+        changed frame => {
+            debug(self.frame);
```
Comment (ogoffart) [inline examples/sprite_sheet/demo.slint:56]: leftover debug

**Comment by ogoffart** [comment]
> This brings back the frame skip when the ball hits the edge of the screen.

Ah right how about:
```
frame-base = self.frame + (travel-direction == TravelDirection.LEFT ? 1 : -1) * animation-tick() / 16ms;
travel-direction = ...
```

**Pull request merged**